sklearn/preprocessing/_function_transformer.py

```diff
@@ -154,9 +154,15 @@ def _check_input(self, X, *, reset):
 
     def _check_inverse_transform(self, X):
         """Check that func and inverse_func are the inverse."""
+        X = np.asarray(X) if not hasattr(X, "dtype") else X
```

Comment:
This is backward incompatible because `transform` (which calls `self.func`) can be designed to take in DataFrame and calling `np.asarray(X)` would convert the DataFrame into an ndarray.

(I am assuming that `validate` is `False`)